dind/pkg/util/util.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+)
+
+// Version reads the version from the metadata stored in the container.
+func DockerImageVersion() (string, error) {
+	f, err := os.Open("/docker_version")
```

Comment:
why do we need this?

Reply:
1. Because the node image is built in a higher level of docker, I give it a unique tag to avoid race conditions with multiple builds.
1. Rather than trying to magically discover the docker tag of the node image, I just write it into a text file.

In the old bazel version, none of this was necessary.